### PR TITLE
Explorer plugin

### PR DESCRIPTION
**Pull request by mostafa-rz**
Issue #4

I test it and it is working now. The only thing is that the React version must be downgraded to 17 based on this [example](https://github.com/graphql/graphiql/blob/HEAD/packages/graphiql-plugin-explorer/examples/index.html). I tried to do it with version 18 but some errors were raised.

### CODE REVIEW

File: src/DownloadAssetsCommand.php [hunk 1]
```diff
@@ -7,14 +7,17 @@
...
 
     public const REACT_DOM_PATH_LOCAL = 'vendor/graphiql/react-dom.production.min.js';
-    public const REACT_DOM_PATH_CDN = '//unpkg.com/react-dom@18/umd/react-dom.production.min.js';
+    public const REACT_DOM_PATH_CDN = '//unpkg.com/react-dom@17/umd/react-dom.production.min.js';
 
     public const JS_PATH_LOCAL = 'vendor/graphiql/graphiql.min.js';
     public const JS_PATH_CDN = '//unpkg.com/graphiql/graphiql.min.js';
 
+    public const EXPLORER_PLUGIN_PATH_LOCAL = 'vendor/graphiql/graphiql-plugin-explorer.umd.js';
+    public const EXPLORER_PLUGIN_PATH_CDN = '//unpkg.com/@graphiql/plugin-explorer@0.1.12/dist/graphiql-plugin-explorer.umd.js';
```
Comment (spawnia) [inline src/DownloadAssetsCommand.php:19]: Given we do not define a version for GraphiQL itself, perhaps we should omit it here too?

File: views/index.blade.php [hunk 2]
```diff
@@ -28,14 +28,37 @@
...
+
+    function GraphiQLWithExplorer() {
+        var [query, setQuery] = React.useState(
+            '',
+        );
+
+        var explorerPlugin = GraphiQLPluginExplorer.useExplorerPlugin({
+            query: query,
+            onEdit: setQuery,
+        });
```
Comment (spawnia) [inline views/index.blade.php:46]: Is this necessary or can we just simplify?

```suggestion
        var explorerPlugin = GraphiQLPluginExplorer.useExplorerPlugin();
```

File: views/index.blade.php [hunk 3]
```diff
@@ -28,14 +28,37 @@
...
+
+        return React.createElement(GraphiQL, {
+            fetcher: fetcher,
+            defaultEditorToolsVisibility: true,
+            plugins: [explorerPlugin],
+            query: query,
+            onEditQuery: setQuery,
+            headerEditorEnabled: true,
+            shouldPersistHeaders: true,
+            defaultHeaders: '{ "Authorization": "Bearer <ENTER YOUR API KEY>" }'
```
Comment (spawnia) [inline views/index.blade.php:56]: Those settings seem orthogonal to adding the plugin, please change only what is strictly necessary

File: views/index.blade.php [hunk 4]
```diff
@@ -28,14 +28,37 @@
 
 <div id="graphiql">Loading...</div>
 <script src="{{ \MLL\GraphiQL\DownloadAssetsCommand::jsPath() }}"></script>
+<script src="{{ \MLL\GraphiQL\DownloadAssetsCommand::explorerPluginPath() }}"></script>
 <script>
+    var fetcher = GraphiQL.createFetcher({
```
Comment (spawnia) [inline views/index.blade.php:33]:
```suggestion
    const fetcher = GraphiQL.createFetcher({
```

I doubt any developer will use a browser not supporting const

File: views/index.blade.php (hunk 2 above)
Comment (mostafa-rz) [reply views/index.blade.php:46]: It seems that it needs them, so it can edit the query in the playground.

Comment (spawnia) [reply views/index.blade.php:46]: Ah I see that they are used in two places - got it!

**Comment by spawnia** [comment]
Please check if everything still works now.

**Comment by mostafa-rz** [comment]
Everything is fine.
